src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -1332,17 +1341,29 @@ public void testDefaultLoggerClosesItStreams() throws Exception {
 
             verify(checker, tmpFile.getPath(), expected);
 
-            assertEquals("Close count was not expected",
-                    1, testInfoOutputStream.getCloseCount());
-            assertEquals("Flush count was not expected",
-                    3, testInfoOutputStream.getFlushCount());
-            assertEquals("Close count was not expected",
-                    1, testErrorOutputStream.getCloseCount());
-            assertEquals("Flush count was not expected",
-                    1, testErrorOutputStream.getFlushCount());
+            assertWithMessage("Output steam close count")
+                    .that(testInfoOutputStream.getCloseCount())
+                    .isEqualTo(1);
+            assertWithMessage("Output steam flush count")
+                    .that(testInfoOutputStream.getFlushCount())
+                    .isAnyOf(3, 4);
+            assertWithMessage("Error steam close count")
+                    .that(testErrorOutputStream.getCloseCount())
+                    .isEqualTo(1);
+            assertWithMessage("Error steam flush count")
+                    .that(testErrorOutputStream.getFlushCount())
+                    .isAnyOf(1, 2);
```

Comment:
weak validation .... sound like ... it can happen or not .

as it depends on jdk version ... can we use something like https://stackoverflow.com/a/2591122/1015848 and be more strict.
Eventually we make minimal jdk as above 13, and test keep be exact.

Reply:
That depends not only on JDK version, it also depends on JDK vendor.
Originally it comes from JDK-13, but was backported to JDK-11 by Apple.

Reply:
Now i get it. It wasn't backported, it is JDK-13 on mac. In that case, you re right, it's easier to check the JDK version.

Reply:
Done.